presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -185,6 +186,28 @@ public int getQueuedQueries()
         }
     }
 
+    @Managed
+    public int getBlockedQueuedQueries()
+    {
+        synchronized (root) {
+            if (canRunMore()) {
+                return 0;
+            }
+            // For leaf group, when it's blocked, all queued queries are blocked on this resource group.
+            if (subGroups.isEmpty()) {
+                return queuedQueries.size();
+            }
+            // For internal groups, when it's blocked, only queries that could run on its subgroups are blocked on this group.
+            int blockedQueuedQUeries = 0;
+            for (InternalResourceGroup subGroup : subGroups.values()) {
+                if (subGroup.canRunMore()) {
```

Comment:
In my head I imagined that this would be a recursive call. Am I missing something?